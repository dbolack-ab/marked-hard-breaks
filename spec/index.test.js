import { marked } from 'marked';
import hardBreaks from '../src/index.js';

function trimLines(s) {
  return s.split('\n').map(l => l.trim()).join('\n');
}

describe('Hard Breaks', ()=>{
  test('Single Break', function() {
    marked.use(hardBreaks());
    expect(marked(trimLines(':\n\n'))).toMatchSnapshot();
  });

  test('Double Break', function() {
    marked.use(hardBreaks());
    expect(marked(trimLines('::\n\n'))).toMatchSnapshot();
  });

  test('Triple Break', function() {
    marked.use(hardBreaks());
    expect(marked(trimLines(':::\n\n'))).toMatchSnapshot();
  });

  test('Many Break', function() {
    marked.use(hardBreaks());
    expect(marked(trimLines('::::::::::\n\n'))).toMatchSnapshot();
  });

  test('Multiple sets of Breaks', function() {
    marked.use(hardBreaks());
    expect(marked(trimLines(':::\n:::\n:::'))).toMatchSnapshot();
  });

  test('Break directly between two paragraphs', function() {
    marked.use(hardBreaks());
    expect(marked(trimLines('Line 1\n::\nLine 2'))).toMatchSnapshot();
  });

  test('Ignored inside a code block', function() {
    marked.use(hardBreaks());
    expect(marked(trimLines('```\n\n:\n\n```\n'))).toMatchSnapshot();
  });
});
