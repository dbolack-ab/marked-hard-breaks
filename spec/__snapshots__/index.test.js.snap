// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hard Breaks Break directly between two paragraphs 1`] = `
"<p>Line 1</p>
<div class='blank'></div><div class='blank'></div>
<p>Line 2</p>
"
`;

exports[`Hard Breaks Double Break 1`] = `
"<div class='blank'></div><div class='blank'></div>
"
`;

exports[`Hard Breaks Ignored inside a code block 1`] = `
"<pre><code>
:
</code></pre>
"
`;

exports[`Hard Breaks Many Break 1`] = `
"<div class='blank'></div><div class='blank'></div><div class='blank'></div><div class='blank'></div><div class='blank'></div><div class='blank'></div><div class='blank'></div><div class='blank'></div><div class='blank'></div><div class='blank'></div>
"
`;

exports[`Hard Breaks Multiple sets of Breaks 1`] = `
"<div class='blank'></div><div class='blank'></div><div class='blank'></div>
<div class='blank'></div><div class='blank'></div><div class='blank'></div>
<div class='blank'></div><div class='blank'></div><div class='blank'></div>
"
`;

exports[`Hard Breaks Single Break 1`] = `
"<div class='blank'></div>
"
`;

exports[`Hard Breaks Triple Break 1`] = `
"<div class='blank'></div><div class='blank'></div><div class='blank'></div>
"
`;
